src/averageAges.js

```diff
@@ -55,7 +65,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const isOnlySon = onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+      && person.mother)
+    : people.filter(person => person.mother);
+
+  const ageDifferences = isOnlySon.map(baby => {
+    const mother = people.find(person => person.name === baby.mother);
+
+    if (mother && baby.born && mother.born) {
```

Comment:
You can simplify the condition by removing 'baby.born && mother.born' since they are always true